tools/c-parser/umm_heap/Arrays.thy

```diff
@@ -273,11 +273,14 @@ next
   show "set_array \<circ> map_array f = (`) f \<circ> set_array"
     by (rule map_array_set_img)
 next
-  show "card_order (BNF_Cardinal_Arithmetic.csum natLeq (card_of UNIV))"
-    by (simp add: card_of_card_order_on card_order_csum natLeq_card_order)
+  show "card_order (BNF_Cardinal_Arithmetic.csum natLeq (card_suc (card_of UNIV)))"
```

Comment:
can you comment on why we went to `card_of ()` to `card_suc (card_of ())`?

Reply:
I have to admit that this change is mostly based off similar changes to BNF declarations in HOL, but the relevant NEWS item is
```
BNFs now require a strict cardinality bound (<o instead of \<le>o).
    Minor INCOMPATIBILITY for existing manual BNF declarations.
```

Reply:
Thanks. I don't really know much about BNF declarations in Isabelle. All good.